### PR TITLE
PYSCAN-2: Add tests for the args parsing.

### PR DESCRIPTION
**Pull request by jdodinh**
Add test for jvm argument parsing.

Make passing of arguments clear.

Revert usage of sys.argv.

### CODE REVIEW

File: src/py_sonar_scanner/__main__.py [hunk 1]
```diff
@@ -17,6 +17,8 @@
 # along with this program; if not, write to the Free Software Foundation,
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #
+import sys
```
Comment (guillaume-dequenne-sonarsource) [inline src/py_sonar_scanner/__main__.py:20]: Is this actually needed?

File: src/py_sonar_scanner/configuration.py [hunk 2]
```diff
@@ -37,43 +37,14 @@ def __init__(self):
         self.sonar_scanner_executable_path = ""
```
Comment (guillaume-dequenne-sonarsource) [inline src/py_sonar_scanner/configuration.py:37]: I cannot highlight the exact line, but the `argparse` import on line 21 is no longer needed.

File: tests/test_configuration.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+import unittest
+from unittest.mock import patch, Mock
+from py_sonar_scanner.configuration import Configuration
+
+
+class TestConfiguration(unittest.TestCase):
+    @patch("py_sonar_scanner.configuration.sys")
+    def test_argument_parsing_empty_toml(self, mock_sys):
```
Comment (guillaume-dequenne-sonarsource) [inline tests/test_configuration.py:27]: One case that would be worth adding is the one where no extra argument is provided.

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
Nice cleanup! I think this simplifies the code quite well and removes some over-engineered bits from the PoC.

It's likely we're going to reintroduce some arg parsing logic eventually if we want to define more Pythonic aliases to the JVM arguments, but as it's not in scope yet I really appreciate the simplified logic.

I left some minor comments on redundant imports and testing suggestion.

File: tests/test_configuration.py (hunk 3 above)
Comment (jdodinh) [reply tests/test_configuration.py:27]: Yes, this makes sense. I added this in the last commit.

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**